code/modules/integrated_electronics/core/helpers.dm

```diff
@@ -0,0 +1,141 @@
+/obj/item/integrated_circuit/proc/setup_io(var/list/io_list, var/io_type, var/list/io_default_list)
+	var/list/io_list_copy = io_list.Copy()
+	io_list.Cut()
+	var/i = 0
+	for(var/io_entry in io_list_copy)
+		i += 1
```

Comment:
The `in` keyword in loops when used like above makes a copy of the list already. For cleaner syntax do:

```
for(var/i in 1 to io_list_copy.len)
   var/io_entry = io_list_copy[i]
```